ext/html_tokenizer_ext/tokenizer.c

```diff
@@ -662,11 +662,30 @@ void tokenizer_scan_all(struct tokenizer_t *tk)
   return;
 }
 
+void tokenizer_set_scan_string(struct tokenizer_t *tk, const char *string, long unsigned int length)
+{
+  const char *old = tk->scan.string;
+  REALLOC_N(tk->scan.string, char, string ? length + 1 : 0);
+  DBG_PRINT("tk=%p realloc(tk->scan.string) %p -> %p length=%lu", tk, old,
+    tk->scan.string, length + 1);
+  if(string && length > 0) {
+    strncpy(tk->scan.string, string, length);
+    tk->scan.string[length] = 0;
```

Comment:
added this to ensure null-terminated `tk->scan.string` regardless of what the original string contains